pkg/cache/application.go

```diff
@@ -166,9 +167,16 @@ func (app *Application) GetSchedulingParamsDefinition() string {
 func (app *Application) setTaskGroups(taskGroups []TaskGroup) {
 	app.lock.Lock()
 	defer app.lock.Unlock()
-	app.taskGroups = taskGroups
-	for _, taskGroup := range app.taskGroups {
-		app.placeholderAsk = common.Add(app.placeholderAsk, common.GetTGResource(taskGroup.MinResource, int64(taskGroup.MinMember)))
+	for _, tg := range taskGroups {
+		taskGroup := tg
+		if _, exists := app.taskGroups[taskGroup.Name]; exists {
+			log.Log(log.ShimCacheApplication).Warn("duplicate task-group within the task-groups",
+				zap.String("appID", app.applicationID),
+				zap.String("groupName", taskGroup.Name))
```

Comment:
Make sure you have an unit test which covers this branch.

Reply:
Well yeah, that 's a good addition. I will provide the unit test which covers this branch.

Reply:
> Make sure you have an unit test which covers this branch.

Well yeah, that 's a good addition. I will provide the unit test which covers this branch.